shared/src/ledger/queries/shell.rs

```diff
@@ -374,12 +375,28 @@ where
                      height",
                 ),
         );
-
+        // from the hashes of the transfers, get the actual values.
+        let keys: Vec<_> =
+            transfer_hashes.iter().map(get_key_from_hash).collect();
+        let values: Vec<PendingTransfer> = keys
+            .iter()
+            .filter_map(|key| match ctx.storage.read(key) {
+                Ok((Some(bytes), _)) => {
+                    BorshDeserialize::try_from_slice(&bytes[..]).ok()
+                }
+                _ => None,
+            })
+            .collect();
+        if values.len() != keys.len() {
+            return Err(storage_api::Error::SimpleMessage(
+                "One or more of the provided hashes had no corresponding \
+                 transfer in storage.",
+            ));
+        }
```

Comment:
we should revisit this later, it should be possible to forward a more specific error message to the caller that would be helpful when debugging (e.g. which hash(es) were missing transfers in storage)

Reply:
I added this.